### DISCUSSION TITLE
Integration with LangChain4j

### DISCUSSION THREAD

**Discussion by ruizrube** [langfuse/langfuse#3256] [Ideas]
### Describe the feature or potential improvement

Hi,

Langfuse currently supports LangChain, Haystack, and LlamaIndex, among other frameworks. All of them are compatible with Python and JavaScript. LangChain is gaining attention for incorporating GenAI features in Java applications, so it would be interesting to develop an integration with this framework.

Regards,

**Comment by marcklingen** [comment]
If I remember correctly, LangChain4j project aims to support otel semantic conventions. Let's track this here. Once they support either callbacks (like langchain) or otel, we could spec an integration together

**Comment by ruizrube** [comment]
They have made some progress in that regard

https://github.com/langchain4j/langchain4j/pull/1058

**Reply by kpavlov** [reply]
I am working on this https://github.com/langchain4j/langchain4j/discussions/2014

**Reply by marcklingen** [reply]
awesome, let me know if you have any questions especially with regards to prompt management. The best resources are the prompt management documentation page and the api reference.

**Reply by stanska** [reply]
Hi,
I will be really happy to use that feature,  do you think it can be available in the near future? Isn't https://github.com/langchain4j/langchain4j/issues/2328 related?
Thank you.

**Reply by marcklingen** [reply]
yes, this is the best thread tracking this: https://github.com/langchain4j/langchain4j/issues/2328
